### PR TITLE
Use `pnpm` instead of `yarn`

### PR DESCRIPTION
**Pull request by leofeyer**
I‘m still a fan of Yarn Berry and its innovative approach, but unfortunately the IDE support seems to be more consistent when there is a `node_modules` folder. I know about the `nodeLinker` option, but that kind of defeats the point of using Yarn 4 IMHO.  Also, after merging this PR, enabling Corepack is no longer mandatory.

### CODE REVIEW

**Comment by bytehead** [comment]
What about to add a `.nvmrc` file with the correct node version we use?

**Comment by leofeyer** [comment]
Does the Node version matter as long as the dependencies are installable?

**Comment by bytehead** [comment]
Yes (as I've experienced it). It can be installable under different versions, but that does not guarantee it's execution.

**Comment by leofeyer** [comment]
I just tried it with Node 18.20.4, 20.15.1 and 22.4.0. Both `pnpm install` and `pnpm build` ran successfully, so I don‘t think we should limit Node to a specific version at the moment.

**Comment by leofeyer** [comment]
Something like `>=18.12` would be helpful, but this does not seem to be supported: https://github.com/nvm-sh/nvm/issues/3167

File: package.json [hunk 1]
```diff
@@ -17,6 +17,5 @@
     "scripts": {
         "build": "encore production --progress",
         "watch": "encore dev --watch"
-    },
-    "packageManager": "yarn@4.1.1"
+    }
```
Comment (aschempp) [inline package.json]: I think we should still specify the package manager we use. So `corepack` will automatically select the correct version for me locally.

Comment (leofeyer) [reply package.json]: Yes, that makes sense. Added again in 83f4b980028ab12b65b7e5f6d89f9518df41c75c.

**Comment by bytehead** [comment]
Then I would prefer to have a `.nvmrc` with the lowest version possible. It's just convenience to run `npm use` before any other `npm` command (imho). Or to look at the file to get an idea which version should be used for running this.

**Comment by leofeyer** [comment]
That does not work, because then I cannot use any other Node version anymore:

**Comment by bytehead** [comment]
Then annotate the version you use.

**Comment by leofeyer** [comment]
Done in a6d2e84fe3bb91e373214a06da2d69699f1610bb.

**Pull request merged**